src/rest_vol_dataset.c

```diff
@@ -1709,10 +1711,29 @@ RV_parse_dataset_creation_properties_callback(char *HTTP_response, void *callbac
      *                                                                            *
      ******************************************************************************/
     if ((key_obj = yajl_tree_get(creation_properties_obj, fill_value_keys, yajl_t_any))) {
-        /* TODO: Until fill value support is implemented, just push an error to the stack but continue ahead
-         */
-        FUNC_DONE_ERROR(H5E_DATASET, H5E_UNSUPPORTED, SUCCEED,
-                        "warning: dataset fill values are unsupported");
+        hid_t  fill_type               = H5I_INVALID_HID;
```

Comment:
`fill_type` should be moved to the top of the function scope and then closed at the end of this function with `H5Tclose` if `fill_type != H5I_INVALID_HID`.